crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -96,6 +96,9 @@ pub struct AuthorityPerpetualTables {
 
     /// Parameters of the system fixed at the epoch start
     pub(crate) epoch_start_configuration: DBMap<(), EpochStartConfiguration>,
+
+    /// A singleton table that stores latest pruned checkpoint. Used to keep objects pruner progress
+    pub(crate) pruned_checkpoint: DBMap<(), CheckpointSequenceNumber>,
```

Comment:
why is the move to this db needed? Is it because we need to atomically update this value in a single write batch?

Reply:
yes, ref counts decrements in the pruner are not idempotent. It needs to be all-or-nothing now
_I don't like having a separate table just for the pruner progress either. Maybe create a general table for singletons?_